.github/actions/scan-docker-image/action.yml

```diff
@@ -0,0 +1,27 @@
+name: Trivy Scan Action
+description: Scan Docker images using Trivy and upload results.
+inputs:
+  image:
+    description: A docker image
+    required: true
+  category:
+    description: Unique category for the SARIF upload
+    required: true
+
+runs:
+  using: composite
+  steps:
+  - name: Run Trivy vulnerability scanner
+    uses: aquasecurity/trivy-action@0.28.0
+    with:
+      image-ref: ${{ inputs.image }}
+      format: sarif
+      output: trivy-results-${{ inputs.image }}.sarif
```

Comment:
`image` is a URL contains `/`, not sure that creating sub directory is ideal for this case; try without the prefix as each job runs one by one?